### PR TITLE
Use cucumber parent

### PR DESCRIPTION
**Pull request by mlvandijk**
## Summary

Use cucumber-parent pom and remove duplication.

## Details

Open issues:
- [x] release updated cucumber-parent and use new version (1.0.2)
- [x] ~~please ensure that all projects or any sub-modules that use jar packaging have project.Automatic-Module-Name set.~~ Update: Cucumber JVM has split packages. So we cant use automatic module names yet. So that needs a work around. Can you disable it for cucumber JVM and keep it for the others? (DONE; added override)
- [x] check config for nexus-staging-maven-plugin (DONE; seems fine)

## Motivation and Context

See  #1493

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mlvandijk** [comment]
Differences in `nexus-staging-maven-plugin`:

**Before using cucumber-parent pom:**
```
      <pluginManagement>

        <plugins>
          <plugin>
            <groupId>org.sonatype.plugins</groupId>
            <artifactId>nexus-staging-maven-plugin</artifactId>
            <version>1.6.7</version>
          </plugin>
        </plugins>
      </pluginManagement>
```

**After adding cucumber-parent pom:**
```
      <pluginManagement>
        <plugins>
            <groupId>org.sonatype.plugins</groupId>
            <artifactId>nexus-staging-maven-plugin</artifactId>
            <version>1.6.7</version>
            <extensions>true</extensions>
            <configuration>
              <serverId>ossrh</serverId>
              <nexusUrl>https://oss.sonatype.org/</nexusUrl>
              <autoReleaseAfterClose>true</autoReleaseAfterClose>
            </configuration>
          </plugin>
        </plugins>
      </pluginManagement>
```

**Comment by mpkorstanje** [comment]
Does it come from the parent? Do we have any other source for the same settings in this pom or are these defaults?

**Comment by mlvandijk** [comment]
Yes, it comes from the parent, see [line 344](https://github.com/cucumber/cucumber-parent/blob/master/pom.xml#L344)

**Comment by mpkorstanje** [comment]
Are they defaults? If so, they can be removed from the pom

**Comment by mpkorstanje** [comment]
From parent pom

**Comment by mlvandijk** [comment]
From the [Maven documentation](http://maven.apache.org/pom.html#Plugin_Management): "pluginManagement: is an element that is seen along side plugins. Plugin Management contains plugin elements in much the same way, except that rather than configuring plugin information for this particular project build, it is intended to configure project builds that inherit from this one. However, this only configures plugins that are actually referenced within the plugins element in the children or in the current POM. The children have every right to override pluginManagement definitions."

Since we use the same configuration in the plugin as in the pluginmanagement, this seems fine to me. If anything we could remove the additional <plugins><plugin> configuration in the cucumber-jvm pom, as it's already inherited from the cucumber-parent. (Have tried locally; this has no effect on the effective pom)

**Comment by mlvandijk** [comment]
Added override for Automatic-Module-Name as suggested [here](https://stackoverflow.com/questions/26736876/difference-between-plugins-and-pluginmanagement-tag-in-maven-pom-xml/26739966).

File: pom.xml [hunk 1]
```diff
@@ -885,16 +788,7 @@
...
-                    <autoVersionSubmodules>true</autoVersionSubmodules>
-                    <useReleaseProfile>false</useReleaseProfile>
-                    <releaseProfiles>release</releaseProfiles>
-                    <goals>deploy</goals>
-                </configuration>
-            </plugin>
+
             <plugin>
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
```
Comment (mpkorstanje) [inline pom.xml]: If this plugin with this configuration is present in the parent, we don't need it here.

Comment (mlvandijk) [reply pom.xml]: I'll double check at home before (squash and) merging.

**Pull request merged**